yarn-project/sequencer-client/src/sequencer/public_processor.ts

```diff
@@ -41,7 +46,14 @@ export class PublicProcessorFactory {
     const publicContractsDB = new ContractsDataSourcePublicDB(this.contractDataSource);
     const worldStatePublicDB = new WorldStatePublicDB(this.merkleTree);
     const worldStateDB = new WorldStateDB(this.merkleTree, this.l1Tol2MessagesDataSource);
-    const publicExecutor = new PublicExecutor(worldStatePublicDB, publicContractsDB, worldStateDB, historicalHeader);
+    const nullifiersDB = new NullifiersDB(this.merkleTree);
```

Comment:
Having this many different DB definitions with two called world state is something we need to take care of. There should be one db that has references to all of these?